### PR TITLE
feat: Add support for toggling between manual/automatic matcher

### PR DESCRIPTION
**Pull request by e00dan**

### CODE REVIEW

File: packages/market-matcher/src/EntityFetcher.ts [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+    getDemandListLength() {
+        return Demand.getDemandListLength(this.config);
+    }
+
+    getSupplyListLength() {
+        return Supply.getSupplyListLength(this.config);
+    }
+
+    getAgreement(id: string, tries = 1) {
```
Comment (kosecki123) [inline packages/market-matcher/src/EntityFetcher.ts]: Previously the default was 10. FYI 1 retry means 1 failed and 1 immediately after. This is our "hack" for awaiting db and blockchain be in sync. I recommend setting it to higher value, perhaps 5-10.

File: packages/market-matcher/src/test/unit/EntityStore.test.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+        const config = Substitute.for<Configuration.Entity>();
+        const logger = Substitute.for<Winston.Logger>();
+        const fetcher = Substitute.for<IEntityFetcher>();
+
+        fetcher.getAgreementListLength().returns(Promise.resolve(0));
+        fetcher.getSupplyListLength().returns(Promise.resolve(0));
+        fetcher.getCertificateListLength().returns(Promise.resolve(0));
+        fetcher.getDemandListLength().returns(Promise.resolve(1));
+        fetcher.getDemand(Arg.all()).returns(Promise.resolve(demand));
```
Comment (kosecki123) [inline packages/market-matcher/src/test/unit/EntityStore.test.ts:32]: Consider moving this block to a separate function, repeated 2 more times below.

**Review by kosecki123 — CHANGES_REQUESTED** [review]
Please take a look at my comments.

File: packages/market-matcher/src/EntityFetcher.ts (hunk 1 above)
Comment (e00dan) [reply packages/market-matcher/src/EntityFetcher.ts]: Set to 10

**Review by josipbagaric — APPROVED** [review]

**Review by kosecki123 — APPROVED** [review]

**Pull request merged**